src/Integration/Binding/BindingProcessImpl.cs

```diff
@@ -122,8 +127,10 @@ public async Task<bool> DownloadQualityProfileAsync(IProgress<FixedStepsProgress
                 }
                 this.InternalState.QualityProfiles[language] = qualityProfileInfo;
 
+                var bindingConfiguration = WriteBindingInformation();
```

Comment:
The location of this calculation sucks. Logically it needs to be done after fetching the quality profile, and before calling GetConfigurationAsync.